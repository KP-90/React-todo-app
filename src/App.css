.App {
    width: 100vw;
    height: 100vh;
    display: grid;
    grid-template-columns: 150px 1fr;
    grid-template-rows: 75px 1fr;
}



.panel {
    background-color: darkgoldenrod;
}

span {
    display: grid;
    grid-template-columns: 2fr 1fr;
}
span > * {
    width: 200px;
}

#sortPriority {
    display: flex;
}

#up, #down {
    width: 15px;
}
#up:hover, #down:hover {
    cursor: pointer;
}
