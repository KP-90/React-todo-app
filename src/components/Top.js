// Handles anything related to the banner/top portion of this app
import './top.css'

const Top = () => {
    return(
        <div className="top">
            <h1>To Do List</h1>
            <button id="add-task">+Add Task</button>
        </div>
    )
}

export default Top