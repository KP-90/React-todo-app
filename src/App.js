import './App.css';
import Top from './components/Top';
import Panel from './components/Panel';
import Main from './components/Main';
import Modal from './components/Modal';

import { useEffect, useState } from 'react';

function App() {
  const [tasks, setTasks] = useState([])
  const [loaded, setLoaded] = useState(false)
  
  const [modalDisplay, setModalDisplay] = useState("none")
  
  //Playing with local storage
  
    if(loaded === false){
      if(localStorage.getItem("tasks")) {
        let saved = localStorage.getItem("tasks")
        console.log("local storage:", JSON.parse(saved))
        setTasks(JSON.parse(saved))
        setLoaded(true)
      }
      else {
        console.log("Setting up local storage")
        let defaultTask = {id:0, task: "Hello, welcome to the App!", priority: "1-High"}
        setTasks([].concat(defaultTask))
        setLoaded(true)
      }
    }
  

  useEffect(() => {
    localStorage.setItem("tasks", JSON.stringify(tasks))
  }, [tasks])
  
  //--Local storage end

  useEffect(() => {
    const handleClick = (e) => {
      const modal = document.querySelector("#newTaskModal")
      setModalDisplay('flex') 
    }
    let btn = document.querySelector("#add-task")
    btn.addEventListener("click", handleClick)
    return() => {btn.removeEventListener("click", handleClick)}
  }, [])

  useEffect(() => {
    window.onclick = (e) => {
      if (e.target !== document.querySelector('.modal-content') && e.target === document.querySelector("#newTaskModal")) {
        setModalDisplay('none')
    }  
    }
  })

  return (
    <div className="App">
      <Modal tasks={tasks} setTasks={setTasks} modalDisplay={modalDisplay} setModalDisplay={setModalDisplay}/>

      <Top />
      <Panel />
      <Main tasks={tasks} setTasks={setTasks}/>
    </div>
  );
}

export default App;
