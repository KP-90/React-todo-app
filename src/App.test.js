import { getByRole, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';
import Top from './components/Top';

describe("App renders", () => {
  it("Renders App", () => {
    const {container} = render(<App />)
    expect(container).toMatchSnapshot()
  })
  it("Renders top section", () => {
    const { getByRole } = render(<Top />)
    expect(getByRole("heading").textContent).toMatch(/to do list/i)
  })
})

describe("Users can add task", () => {
  beforeEach(() => {
    const {container} = render(<App />)
  })
  it("Button show modal", () => {
    const btn = screen.getByRole("button", {name: "+Add Task"})
    userEvent.click(btn)
    expect(screen.getByRole("modal").style.display).toMatch(/flex/i)

  })
})