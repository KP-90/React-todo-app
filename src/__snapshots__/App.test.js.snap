// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders Renders App 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="modal"
      id="newTaskModal"
      role="modal"
      style="display: none;"
    >
      <div
        class="modal-content"
      >
        <form>
          <label>
            Task:
            <input
              role="inputText"
              type="text"
            />
          </label>
          <label>
            Priority:
            <select>
              <option
                value="1-High"
              >
                1-High
              </option>
              <option
                value="2-Normal"
              >
                2-Normal
              </option>
              <option
                value="3-Low"
              >
                3-Low
              </option>
            </select>
          </label>
        </form>
        <button
          id="close"
          role="close"
        >
          Submit & Close
        </button>
      </div>
    </div>
    <div
      class="top"
    >
      <h1>
        To Do List
      </h1>
      <button
        id="add-task"
      >
        +Add Task
      </button>
    </div>
    <div
      class="panel"
    />
    <div
      class="main"
    >
      <span>
        <h2>
          Task
        </h2>
        <h2
          id="sortPriority"
        >
          <span
            class="sorting"
            id="up"
          >
            ↑
          </span>
          Priority
          <span
            class="sorting"
            id="down"
          >
            ↓
          </span>
        </h2>
      </span>
      <div
        id="0"
        role="task-container"
      >
        <div
          class="modal"
          id="taskinfo"
          style="display: none;"
        >
          <div
            class="modal-content"
            id="taskInfoContent"
          >
            <h1>
              Hello, welcome to the App!
            </h1>
          </div>
        </div>
        <div
          class="task"
        >
          <input
            type="checkbox"
          />
          <div
            class="editDiv"
            hidden=""
          >
            <textarea
              cols="45"
              id="editInput"
              maxlength="180"
            >
              Hello, welcome to the App!
            </textarea>
            <button
              class="done"
            >
              done
            </button>
          </div>
          <p
            class=""
            id="0"
          >
            Hello, welcome to the App!
          </p>
          <p>
            1-High
          </p>
          <button
            class="edit"
          >
            Edit
          </button>
          <button
            class="delete"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
